src/RouterInjector.ts

```diff
@@ -0,0 +1,88 @@
+import { beforeRender } from '@dojo/widget-core/WidgetBase';
+import { w, registry as globalRegistry } from '@dojo/widget-core/d';
+import { WidgetRegistry } from '@dojo/widget-core/WidgetRegistry';
+import { Injector, BaseInjector, InjectorProperties } from '@dojo/widget-core/Injector';
+import { DNode, RegistryLabel } from '@dojo/widget-core/interfaces';
+
+import HashHistory from './history/HashHistory';
+import { History } from './history/interfaces';
+import { Router, RouteConfig } from './Router';
+import { MatchType } from './Route';
+import { MapParamsOptions, OutletProperties } from './interfaces';
+
+/**
+ * Key for the router injetor
+ */
+export const routerKey = Symbol();
+
+/**
+ * Creates a router instance for a specific History manager (default is `HashHistory`) and registers
+ * the route configuration.
+ *
+ * @param config The route config to register for the router
+ * @param registry An optional registry that defaults to the global registry
+ * @param history The history manager the router needs to use, default `HashHistory`
```

Comment:
"default `HashHistory`" -> "default is `HashHistory`" or "defaults to `HashHistory`"